### PR TITLE
Deprecate MenuCustomizations feature

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- Deprecate MenuCustomizations feature

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] PR is ideally < ~400LOC

### CODE REVIEW

File: packages/components/src/Menu/MenuContext.ts [hunk 1]
```diff
@@ -38,18 +37,19 @@ export interface MenuContextProps {
   triggerCallbackRef?: (node: HTMLElement | null) => void
 }
 
-export interface MenuItemStyleContextProps extends MenuSharedProps {
+export interface MenuItemContextProps {
+  compact?: boolean
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuContext.ts:41]: 👍

File: packages/components/src/Menu/MenuGroupLabel.tsx [hunk 2]
```diff
@@ -57,15 +43,13 @@ export const MenuGroupLabel: FC<MenuGroupLabelProps> = ({
       */}
       <div ref={labelShimRef} style={{ height: '0' }} />
       <Heading
-        fontSize={fontSize || 'small'}
+        fontSize="small"
         as="h2"
         px="medium"
         py="xsmall"
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuGroupLabel.tsx:49]: Totally not necessary for this PR but I just remembered a request to also make the `MenuGroupLabel` padding shrink down if `compact` is used.

File: packages/components/src/Menu/MenuItemLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        ref={ref}
+      />
+    )
+  }
+)
+
+MenuItemLayoutInternal.displayName = 'MenuItemLayoutInternal'
+
+const hoverStyles = ({ current }: MenuListItemProps) =>
+  !current &&
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItemLayout.tsx:58]: Also not necessary in this PR but we might think about getting rid of the `current` prop here and just using this approach instead:
```
&[aria-current] {
  // current styles go here
}
&:not([aria-current]) {
  // non-current styles go here
}
```

File: packages/components/src/Menu/MenuItemLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * All of this drama is to deal with SC's behavior of auto-spreading the Element interface
+ * used when styled extends a base type. E.g. (styled.li has `color` prop)
+ */
+const MenuItemLayoutInternal = forwardRef(
+  (props: MenuListItemProps, ref: Ref<HTMLLIElement>) => {
+    return (
+      <li
+        {...omit(props, 'compact', 'current', 'focusVisible', 'hasIcon')}
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItemLayout.tsx:47]:
```suggestion
        {...omit(props, 'compact', 'focusVisible', 'hasIcon')}
```

File: packages/components/src/Menu/MenuItemLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+MenuItemLayoutInternal.displayName = 'MenuItemLayoutInternal'
+
+export const MenuItemLayout = styled(MenuItemLayoutInternal)`
+  align-items: center;
+  color: ${({ theme: { colors } }) => colors.text1};
+  display: flex;
+  flex-wrap: wrap;
+  font-size: ${({ theme: { fontSizes } }) => fontSizes.small};
+  font-weight: ${({ current, theme: { fontWeights } }) =>
+    current ? fontWeights.semiBold : fontWeights.normal};
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItemLayout.tsx]:
```suggestion
  &[aria-current] {
    font-weight: ${({ theme: { fontWeights } }) => fontWeights.semiBold};
  }
```

File: packages/components/src/Menu/MenuDisclosure.tsx [hunk 6]
```diff
@@ -139,8 +139,8 @@ export const MenuDisclosure: FC<MenuDisclosureProps> = ({
 
   return (
     <>
-      {renderedTooltip}
       {cloned}
+      {renderedTooltip}
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuDisclosure.tsx:143]: Better than before but now the tooltip has the left-margin. This is a possible, though awkward solution. Future PR?
```
  && > :not(${Outer}) {
    margin-left: ${({ theme }) => theme.space[space.gap || defaultSpaceSize]};
  }
```

File: packages/components/src/Menu/MenuItemLayout.tsx (hunk 5 above)
Comment (lukelooker) [reply packages/components/src/Menu/MenuItemLayout.tsx]: Already done :)

File: packages/components/src/Menu/MenuDisclosure.tsx (hunk 6 above)
Comment (lukelooker) [reply packages/components/src/Menu/MenuDisclosure.tsx:143]: Yep, need to refine here in general a bit. Will address separately.

**Pull request merged**